solr/test-framework/src/java/org/apache/solr/util/SolrJettyTestRule.java

```diff
@@ -0,0 +1,122 @@
+import static org.apache.solr.SolrTestCaseJ4.writeCoreProperties;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Path;
+import java.util.Properties;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.embedded.JettyConfig;
+import org.apache.solr.embedded.JettySolrRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrJettyTestRule extends SolrClientTestRule {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private JettySolrRunner jetty;
+  private SolrClient adminClient = null;
+
+  @Override
+  protected void after() {
+    if (adminClient != null) {
+      try {
+        adminClient.close();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    adminClient = null;
+
+    if (jetty != null) {
+      try {
+        jetty.stop();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      jetty = null;
+    }
+  }
+
+  public void reset() throws Exception {
+    after();
+  }
+
+  @Override
+  public void startSolr(Path solrHome) {
+    startSolr(
+        solrHome,
+        new Properties(),
+        JettyConfig.builder()
+            .withSSLConfig(SolrTestCaseJ4.sslConfig.buildServerSSLConfig())
+            .build());
+  }
+
+  public void startSolr(Path solrHome, JettyConfig jettyConfig) {
+    startSolr(solrHome, new Properties(), jettyConfig);
+  }
+
+  public void startSolr(Path solrHome, Properties nodeProperties, JettyConfig jettyConfig) {
+
+    try {
+      initCore(null, null, solrHome.toString());
+
+      Path coresDir = createTempDir().resolve("cores");
+
+      Properties props = new Properties();
+      props.setProperty("name", DEFAULT_TEST_CORENAME);
+      props.setProperty("configSet", "collection1");
+      props.setProperty("config", "${solrconfig:solrconfig.xml}");
+      props.setProperty("schema", "${schema:schema.xml}");
+
+      writeCoreProperties(coresDir.resolve("core"), props, "RestTestBase");
+
+      Properties nodeProps = new Properties(nodeProperties);
+      nodeProps.setProperty("coreRootDirectory", coresDir.toString());
+      nodeProps.setProperty("configSetBaseDir", solrHome.toString());
+
+      jetty = new JettySolrRunner(solrHome.toString(), nodeProps, jettyConfig);
+      jetty.start();
+      int port = jetty.getLocalPort();
+      log.info("Jetty Assigned Port#{}", port);
+      adminClient = SolrTestCaseJ4.getHttpSolrClient(jetty.getBaseUrl().toString());
+    } catch (Exception e) {
+      log.error(e.toString());
+    }
+  }
+
+  public JettySolrRunner getJetty() {
+    if (jetty == null) throw new IllegalStateException("Jetty has not started");
+    return jetty;
+  }
+
+  @Override
+  public SolrClient getSolrClient(String name) {
+    throw new UnsupportedOperationException(); // TODO Client needed
```

Comment:
is this still a todo?

Reply:
Its now implemented.` return getHttpSolrClient(jetty.getBaseUrl().toString() + "/" + DEFAULT_TEST_CORENAME);`

Reply:
awesome!